sources/TileDB.CSharp/Enums.cs

```diff
@@ -1105,6 +1099,11 @@ public static VfsMode VfsModeFromStr(string str)
         public static tiledb_datatype_t to_tiledb_datatype(Type t) =>
             (tiledb_datatype_t)TypeToDataType(t);
 
+        /// <summary>
+        /// Converts a <see cref="Type"/> to a <see cref="DataType"/> enum value.
+        /// </summary>
+        /// <param name="t">The type to convert.</param>
+        /// <exception cref="NotSupportedException"><paramref name="t"/> is unsupported.</exception>
         public static DataType TypeToDataType(Type t)
```

Comment:
I don't like this, it restricts us to a one-to-one mapping between .NET types and TileDB data types (it will be a problem once we allow mapping between TileDB and .NET booleans). We might need a `bool IsCompatible<T>(DataType dataType)` method.